src/Farmer/Common.fs

```diff
@@ -594,6 +592,20 @@ module WebApp =
     module Extensions =
         /// The Microsoft.AspNetCore.AzureAppServices logging extension.
         let Logging = ExtensionName "Microsoft.AspNetCore.AzureAppServices.SiteExtension"
+    open Validation
+    type WebAppName =
+        private | WebAppName of ResourceName
+        static member Create name =
```

Comment:
Hmmm, good idea, let me try that out

Reply:
Done! There's now a function that looks as follows:

```fsharp
static member internal Empty = WebAppName ResourceName.Empty
```

Notice that it's internal so only Farmer itself can use this. External callers can only call the standard `Create` method. I like this solution - it feels a good compromise between type safety and death by types.

Now Functions and WebApps both use `WebApp.Empty` instead of the "defaultvalue" stuff. In the `Run` method, we then do a quick check and fail there.